### PR TITLE
LPS-152404 Create CompareDates function

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:sf
ci:test:relevant

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:stop

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/validation/util/DateParameterUtil.java [hunk 1]
```diff
@@ -47,8 +52,21 @@ public static LocalDate getLocalDate(String dateString) {
...
-			dateString, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+		try {
+			SimpleDateFormat simpleDateFormat = new SimpleDateFormat(
+				_DATE_FORMAT);
+
+			return LocalDate.parse(
+				simpleDateFormat.format(
+					DateUtil.parseDate(
+						_DATE_FORMAT, dateString, LocaleUtil.getDefault())),
+				DateTimeFormatter.ofPattern(_DATE_FORMAT));
```
Comment (MarinhoFeliphe) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/validation/util/DateParameterUtil.java:63]: Hi @seltong, let's try to find a better way to convert this date to the pattern date.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by seltong** [comment]
ci:test:sf

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:forward

**Comment by seltong** [comment]
ci:forward

**Comment by seltong** [comment]
ci:forward

**Comment by seltong** [comment]
ci:forward

**Comment by seltong** [comment]
ci:forward

**Comment by seltong** [comment]
ci:forward

**Pull request closed without merging**